src/Events/ScormLaunched.php

```
@@ -8,9 +8,23 @@ class ScormLaunched extends Event {
      * @override Event
      */
     public function read(array $opts) {
+        $parent = parent::read($opts);
+        if (is_null($parent)) {
```

I'd rather throw an exception in all of these cases and catch them where necessary, since I'm not a fan of error values at all.